openmc/universe.py

```diff
@@ -100,10 +100,17 @@ def get_all_universes(self):
             :class:`Universe` instances
 
         """
+
+        if memo and self in memo:
+            return {}
+
+        if memo is not None:
+            memo.add(self)
```

Comment:
This code doesn't seem right to me. In the top, you are returning a dictionary, and in the bottom you are treating it like a set. The type should be consistent.

Reply:
Yes, the `return {}` is the return value of the function, unrelated to the memo type, which is a `set`. Am I missing your meaning?

Reply:
Nope, you're absolutely right... I guess now the weird part to me is that the `memo` argument is not really being used to save the result of the function call based on the inputs but rather just indicating whether the function has been called with the same arguments. That is not what I think of to be traditional "memoization" (basically, a function result cache). As you point out in your other comment though, this is a pre-existing pattern for other objects, so for now it's probably best we stick with that solution.